DoLahTest/ParserUnitTest.cpp

```diff
@@ -379,6 +387,130 @@ namespace DoLahTest {
         }
     }
 
+    // From here, integrated test on special cases of 'toDateFormat' for relative time descriptions.
+    // The time will auto-filled, and the behaviour changes depending on the time of running.
+    // Test will be disabled in a certain condition where it cannot run properly.
+    TEST_METHOD(DateGivenDayAndMonth) {
+        if (month == 1 && day == 1) {
+            Assert::IsTrue(true);
```

Comment:
The expected answer is the next new year in 364 days out of 365 days because the current new year is already in the past, but on the 1st of Jan it will return 1st of Jan of the same year because the new year has not passed.